# <span style="color:blue"> 이토록 쉬운 통계 & R </span>

<img src="../image/cover.png" width="200" height="400" align="left">  임경덕 지음

## PART 4 데이터 분석 도구, R

## 95% R로 분석 다시보기-1-salary-1 (37일차)

   * 데이타프로젝트 (http://dataproject.xyz) 또는 (http://cafe.naver.com/dataartproject) 에 미리 저장된 스크립트를 다운 받을 수 있습니다.
   * 스크립트를 활용하면 직접 명령어를 입력하지 않고 저장된 명령어들을 실행할 수 있습니다.

### 데이터 파일은 dataartproject.xyz 에 있습니다.

# 하나의 연속형 변수 요약하기

## 내 연봉은 어느 정도 순위에 있을까?

한국 납세자 연맹(koreatax.org) 에서는 연말정산 데이터를 기반으로 연봉 탐색 서비스를 제공합니다. 이것을 바탕으로 가상의 기업을 만들어 10,000 명이 근무하는 회사의 연봉 데이터를 만들었습니다.

In [24]:
data = subway = read.csv("Practice/data/salary_data.csv")

In [11]:
head(data)

salary
1 2930  
2 6122  
3 3301  
4  979  
5 4926  
6 2328

In [12]:
nrow(data)

[1] 10000

   * 10,000 개의 관측치가 있습니다.
   * 누군가의 연봉을 말합니다. 단위는 만원입니다.
   * 이제 이 데이타로 요약을 해보아요


In [13]:
salary = data$salary

In [15]:
head(salary, 50)

[1] 2930 6122 3301  979 4926 2328   82 6879  672  350 1812  158 5105 4945  170
[16] 3914 2042 1966 1020 4899 1575 2714 1451  686 1403 3429 7034 3485 3721 3215
[31] 5755 2087  160  276 3208 2855 1055 2648 4901 1377  108 6167 3005  472 2253
[46]  408 3127  386  392 1948

In [16]:
# 오름차순으로 정렬하여 순서매기기를 해봅니다.
head(sort(salary),50)

[1]  0  0  1  1  1  2  2  3  3  3  4  4  4  4  5  6  6  7  7  7  7  8  8  8  9
[26]  9  9 10 10 10 10 10 12 13 13 13 13 14 14 15 15 16 17 17 18 18 18 18 19 20

In [17]:
# 최솟값
min(salary)

[1] 0

In [18]:
#최댓값
max(salary)

[1] 86669

In [19]:
#중간값
median(salary)

[1] 2276

   * 최솟값은 0원, 최댓값은 8억 6천 669만원입니다.
   * 중간값은 2천 276만원입니다.

In [20]:
# 사분위수를 확인 해 보아요
quantile(salary, 0.9)

90% 
6637

In [21]:
quantile(salary, 0.25)

25% 
1194.75

In [22]:
quantile(salary, 0.75)

75% 
4221.25

   * 90% 지점 연봉이 보이고
   * Q1 은 약 1,195 만원
   * Q3 는 4,221 만원입니다.
   
   
   
   * 이것을 한번에 하는 명령어는 summary 입니다.

In [23]:
summary(salary)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
      0    1195    2276    3140    4221   86669 